lib/string.js

```diff
@@ -91,19 +91,30 @@ internals.String = class extends Any {
         });
     }
 
-    regex(pattern, name) {
+    regex(pattern, config) {
 
         Hoek.assert(pattern instanceof RegExp, 'pattern must be a RegExp');
 
         pattern = new RegExp(pattern.source, pattern.ignoreCase ? 'i' : undefined);         // Future version should break this and forbid unsupported regex flags
 
-        return this._test('regex', pattern, function (value, state, options) {
+        const patternIsInvalidated = (typeof config === 'object' && Hoek.reach(config, 'invalidate'));
+        const testName = patternIsInvalidated ? 'invalidatedRegex' : 'regex';
 
-            if (pattern.test(value)) {
+        return this._test(testName, pattern, function (value, state, options) {
```

Comment:
I changed the test name to distinguish between a normal regex test and an inverted regex test in the `describe`'d value. Using the revision you requested, should the name/pattern/inverted flag just become part of the `args` property on the describe?